### PR TITLE
Add Windows Application channel rules

### PR DESCRIPTION
**Pull request by cristgl**
This PR adds some rules for the provider names contained in the channel Application. The provider names included in this ruleset are listed in the issue https://github.com/wazuh/wazuh-ruleset/issues/310

Testing
---------
Alert triggering has been tested on the next systems:

- Windows 2008 R2
- Windows 2012 R2
- Windows 8
- Windows 10
- Windows Server 2019

### CODE REVIEW

**Comment by Skeptor** [comment]
- [x] Check manager is not flooded by Application events
Triggered Application events:
  Alerts:
   · Windows System error event
   · The software protection service was programmed correctly to restart at {date}
  Events without alert:
   · License time with remaining time limit: validity hours X minutes.
   · The software protection service is starting.

File: rules/0575-windows_rules.xml [hunk 1]
```diff
@@ -0,0 +1,5170 @@
...
+  </rule>
+
+  <rule id="20424" level="4">
+    <if_sid>20421</if_sid>
+    <field name="win.system.eventID">^3024$</field>
+    <description>Cannot start update because the content sources cannot be accessed</description>
+    <options>no_full_log</options>
+  </rule>
+
+  <rule id="20425" level="5">
```
Comment (chemamartinez) [inline rules/0575-windows_rules.xml]: If it is a **critical** error it cannot be level 5. At least it should be level 9.

File: rules/0575-windows_rules.xml [hunk 2]
```diff
@@ -0,0 +1,5170 @@
...
+  </rule>
+
+  <rule id="20428" level="3">
+    <if_sid>20421</if_sid>
+    <field name="win.system.eventID">^3052$</field>
+    <description>Crawl requested to be stopped</description>
+    <options>no_full_log</options>
+  </rule>
+
+  <rule id="20429" level="3">
```
Comment (chemamartinez) [inline rules/0575-windows_rules.xml]: Level 7

File: rules/0575-windows_rules.xml [hunk 3]
```diff
@@ -0,0 +1,5170 @@
...
+
+  <rule id="20385" level="5">
+    <if_sid>20382</if_sid>
+    <field name="win.system.eventID">^11707$|^1033$</field>
+    <options>alert_by_email</options>
+    <description>Windows: Application Installed $(win.eventdata.data)</description>
+    <options>no_full_log</options>
+  </rule>
+
+    <rule id="20386" level="3">
```
Comment (chemamartinez) [inline rules/0575-windows_rules.xml]: Level 5. It is an error.

File: rules/0575-windows_rules.xml [hunk 4]
```diff
@@ -0,0 +1,5170 @@
...
+    <if_sid>20382</if_sid>
+    <field name="win.system.eventID">^1001$</field>
+    <description>Request for product component failed</description>
+    <options>no_full_log</options>
+  </rule>
+
+  <rule id="20387" level="3">
```
Comment (chemamartinez) [inline rules/0575-windows_rules.xml]: Level 5.

File: rules/0575-windows_rules.xml [hunk 5]
```diff
@@ -0,0 +1,5170 @@
...
+    <if_sid>20382</if_sid>
+    <field name="win.system.eventID">^1002$</field>
+    <description>Unexpected or missing value in key</description>
+    <options>no_full_log</options>
+  </rule>
+
+  <rule id="20388" level="3">
```
Comment (chemamartinez) [inline rules/0575-windows_rules.xml]: Level 5

File: rules/0575-windows_rules.xml [hunk 6]
```diff
@@ -0,0 +1,5170 @@
...
+  </rule>
+
+  <rule id="20390" level="4">
+    <if_sid>20382</if_sid>
+    <field name="win.system.eventID">^1007$</field>
+    <description>Installation not permitted by software restriction policy</description>
+    <options>no_full_log</options>
+  </rule>
+
+  <rule id="20391" level="3">
```
Comment (chemamartinez) [inline rules/0575-windows_rules.xml]: Level 5

File: rules/0575-windows_rules.xml [hunk 7]
```diff
@@ -0,0 +1,5170 @@
...
+    <if_sid>20382</if_sid>
+    <field name="win.system.eventID">^1008$</field>
+    <description>Installation not permitted due to an error in software restriction policy processing</description>
+    <options>no_full_log</options>
+  </rule>
+
+  <rule id="20392" level="3">
```
Comment (chemamartinez) [inline rules/0575-windows_rules.xml]: Level 5

File: rules/0575-windows_rules.xml [hunk 8]
```diff
@@ -0,0 +1,5170 @@
...
+    <if_sid>20382</if_sid>
+    <field name="win.system.eventID">^1012$</field>
+    <description>64-bit packages deployment is not supported for this version</description>
+    <options>no_full_log</options>
+  </rule>
+
+  <rule id="20393" level="3">
```
Comment (chemamartinez) [inline rules/0575-windows_rules.xml]: Level 5

File: rules/0575-windows_rules.xml [hunk 9]
```diff
@@ -0,0 +1,5170 @@
...
+  </rule>
+
+  <rule id="20394" level="3">
+    <if_sid>20382</if_sid>
+    <field name="win.system.eventID">^1015$</field>
+    <description>MsiInstaller: Failed to connect to server</description>
+    <options>no_full_log</options>
+  </rule>
+
+  <rule id="20395" level="4">
```
Comment (chemamartinez) [inline rules/0575-windows_rules.xml]: Level 5

File: rules/0575-windows_rules.xml [hunk 10]
```diff
@@ -0,0 +1,5170 @@
...
+  </rule>
+
+  <rule id="20393" level="3">
+    <if_sid>20382</if_sid>
+    <field name="win.system.eventID">^1014$</field>
+    <description>Windows Installer proxy information not correctly registered</description>
+    <options>no_full_log</options>
+  </rule>
+
+  <rule id="20394" level="3">
```
Comment (chemamartinez) [inline rules/0575-windows_rules.xml]: Level 5

File: rules/0575-windows_rules.xml [hunk 11]
```diff
@@ -0,0 +1,5170 @@
...
+  </rule>
+
+  <rule id="20395" level="4">
+    <if_sid>20382</if_sid>
+    <field name="win.system.eventID">^1016$</field>
+    <description>Detection of product feature component failed</description>
+    <options>no_full_log</options>
+  </rule>
+
+  <rule id="20396" level="3">
```
Comment (chemamartinez) [inline rules/0575-windows_rules.xml]: Level 5

File: rules/0575-windows_rules.xml [hunk 12]
```diff
@@ -0,0 +1,5170 @@
...
+  </rule>
+
+  <rule id="20397" level="3">
+    <if_sid>20382</if_sid>
+    <field name="win.system.eventID">^1019$</field>
+    <description>Update successfully removed</description>
+    <options>no_full_log</options>
+  </rule>
+
+  <rule id="20398" level="3">
```
Comment (chemamartinez) [inline rules/0575-windows_rules.xml]: Level 5

File: rules/0575-windows_rules.xml [hunk 13]
```diff
@@ -0,0 +1,5170 @@
...
+  </rule>
+
+  <rule id="20399" level="3">
+    <if_sid>20382</if_sid>
+    <field name="win.system.eventID">^1022$</field>
+    <description>Update installed successfully</description>
+    <options>no_full_log</options>
+  </rule>
+
+  <rule id="20400" level="3">
```
Comment (chemamartinez) [inline rules/0575-windows_rules.xml]: Level 5

File: rules/0575-windows_rules.xml [hunk 14]
```diff
@@ -0,0 +1,5170 @@
...
+    <if_sid>20382</if_sid>
+    <field name="win.system.eventID">^1024$</field>
+    <description>Update could not be installed</description>
+    <options>no_full_log</options>
+  </rule>
+
+  <rule id="20401" level="3">
```
Comment (chemamartinez) [inline rules/0575-windows_rules.xml]: Level 5

File: rules/0575-windows_rules.xml [hunk 15]
```diff
@@ -0,0 +1,5170 @@
...
+    <if_sid>20382</if_sid>
+    <field name="win.system.eventID">^1025$</field>
+    <description>File in use by another program</description>
+    <options>no_full_log</options>
+  </rule>
+
+  <rule id="20402" level="3">
```
Comment (chemamartinez) [inline rules/0575-windows_rules.xml]: Level 7

File: rules/0575-windows_rules.xml [hunk 16]
```diff
@@ -0,0 +1,5170 @@
...
+  </rule>
+
+  <rule id="20403" level="3">
+    <if_sid>20382</if_sid>
+    <field name="win.system.eventID">^1029$</field>
+    <description>The installation or update for the product required a restart for all changes to take effect</description>
+    <options>no_full_log</options>
+  </rule>
+
+  <rule id="20404" level="3">
```
Comment (chemamartinez) [inline rules/0575-windows_rules.xml]: Level 5

File: rules/0575-windows_rules.xml [hunk 17]
```diff
@@ -0,0 +1,5170 @@
...
+  </rule>
+
+  <rule id="20405" level="3">
+    <if_sid>20382</if_sid>
+    <field name="win.system.eventID">^1031$</field>
+    <description>Assembly component is in use</description>
+    <options>no_full_log</options>
+  </rule>
+
+  <rule id="20406" level="3">
```
Comment (chemamartinez) [inline rules/0575-windows_rules.xml]: Level 5

File: rules/0575-windows_rules.xml [hunk 18]
```diff
@@ -0,0 +1,5170 @@
...
+  </rule>
+
+  <rule id="20410" level="3">
+    <if_sid>20382</if_sid>
+    <field name="win.system.eventID">^1037$</field>
+    <description>Windows installer removed an update</description>
+    <options>no_full_log</options>
+  </rule>
+
+  <rule id="20411" level="4">
```
Comment (chemamartinez) [inline rules/0575-windows_rules.xml]: Level 0 -> I don't think it is necessary to fire a rule to notice the Windows machine has to be restarted to apply changes. It should be logged in the host locally.

File: rules/0575-windows_rules.xml [hunk 19]
```diff
@@ -0,0 +1,5170 @@
...
+  </rule>
+
+  <rule id="20413" level="0">
+    <if_sid>20001</if_sid>
+    <field name="win.system.providerName">Microsoft-Windows-Security-SPP</field>
+    <description>Group of SPP events</description>
+    <options>no_full_log</options>
+  </rule>
+
+  <rule id="20414" level="3">
```
Comment (chemamartinez) [inline rules/0575-windows_rules.xml]: The Software **Protection** service **stopped** -> Level 5

File: rules/0575-windows_rules.xml [hunk 20]
```diff
@@ -0,0 +1,5170 @@
...
+  </rule>
+
+  <rule id="20418" level="3">
+    <if_sid>20413</if_sid>
+    <field name="win.system.eventID">^8225$</field>
+    <description>The existing scheduler data does not match the expected data</description>
+    <options>no_full_log</options>
+  </rule>
+
+  <rule id="20419" level="3">
```
Comment (chemamartinez) [inline rules/0575-windows_rules.xml]: Level 5

File: rules/0575-windows_rules.xml [hunk 21]
```diff
@@ -0,0 +1,5170 @@
...
+  </rule>
+
+  <rule id="20421" level="0">
+    <if_sid>20001</if_sid>
+    <field name="win.system.providerName">^Microsoft-Windows-Search$</field>
+    <description>Group of Windows Search events</description>
+    <options>no_full_log</options>
+  </rule>
+
+  <rule id="20422" level="3">
```
Comment (chemamartinez) [inline rules/0575-windows_rules.xml]: Level 5

File: rules/0575-windows_rules.xml [hunk 22]
```diff
@@ -0,0 +1,5170 @@
...
+    <if_sid>20421</if_sid>
+    <field name="win.system.eventID">^3007$</field>
+    <description>Performance monitoring cannot be initialized</description>
+    <options>no_full_log</options>
+  </rule>
+
+  <rule id="20423" level="4">
```
Comment (chemamartinez) [inline rules/0575-windows_rules.xml]: Sounds bad -> level 5

File: rules/0575-windows_rules.xml [hunk 23]
```diff
@@ -0,0 +1,5170 @@
...
+    <if_sid>20421</if_sid>
+    <field name="win.system.eventID">^3014$</field>
+    <description>Internal error generated by a component loaded in the system</description>
+    <options>no_full_log</options>
+  </rule>
+
+  <rule id="20424" level="4">
```
Comment (chemamartinez) [inline rules/0575-windows_rules.xml]: Level 5

File: rules/0575-windows_rules.xml [hunk 24]
```diff
@@ -0,0 +1,5170 @@
...
+  </rule>
+
+  <rule id="20425" level="5">
+    <if_sid>20421</if_sid>
+    <field name="win.system.eventID">^3025$</field>
+    <description>Critical error: System low on resources</description>
+    <options>no_full_log</options>
+  </rule>
+
+  <rule id="20426" level="5">
```
Comment (chemamartinez) [inline rules/0575-windows_rules.xml]: Level 7

File: rules/0575-windows_rules.xml [hunk 25]
```diff
@@ -0,0 +1,5170 @@
...
+    <if_sid>20421</if_sid>
+    <field name="win.system.eventID">^3026$</field>
+    <description>Advise Status Change failed: System low on resources</description>
+    <options>no_full_log</options>
+  </rule>
+
+  <rule id="20427" level="3">
```
Comment (chemamartinez) [inline rules/0575-windows_rules.xml]: Level 5

File: rules/0575-windows_rules.xml [hunk 26]
```diff
@@ -0,0 +1,5170 @@
...
+  </rule>
+
+  <rule id="20429" level="3">
+    <if_sid>20421</if_sid>
+    <field name="win.system.eventID">^3085$</field>
+    <description>Application network access account invalid</description>
+    <options>no_full_log</options>
+  </rule>
+
+  <rule id="20430" level="3">
```
Comment (chemamartinez) [inline rules/0575-windows_rules.xml]: Level 5

File: rules/0575-windows_rules.xml [hunk 27]
```diff
@@ -0,0 +1,5170 @@
...
+    <if_sid>20421</if_sid>
+    <field name="win.system.eventID">^3087$</field>
+    <description>Cannot flush gatherer files</description>
+    <options>no_full_log</options>
+  </rule>
+
+  <rule id="20431" level="3">
```
Comment (chemamartinez) [inline rules/0575-windows_rules.xml]: Level 5

File: rules/0575-windows_rules.xml [hunk 28]
```diff
@@ -0,0 +1,5170 @@
...
+    <if_sid>20421</if_sid>
+    <field name="win.system.eventID">^3088$</field>
+    <description>Cannot update checkpoint record</description>
+    <options>no_full_log</options>
+  </rule>
+
+  <rule id="20432" level="3">
```
Comment (chemamartinez) [inline rules/0575-windows_rules.xml]: Level 5

File: rules/0575-windows_rules.xml [hunk 29]
```diff
@@ -0,0 +1,5170 @@
...
+    <if_sid>20421</if_sid>
+    <field name="win.system.eventID">^3089$</field>
+    <description>Cannot save gatherer files</description>
+    <options>no_full_log</options>
+  </rule>
+
+  <rule id="20433" level="3">
```
Comment (chemamartinez) [inline rules/0575-windows_rules.xml]: Level 5

File: rules/0575-windows_rules.xml [hunk 30]
```diff
@@ -0,0 +1,5170 @@
...
+    <if_sid>20421</if_sid>
+    <field name="win.system.eventID">^3090$</field>
+    <description>Cannot restore gatherer files from the previous checkpoint</description>
+    <options>no_full_log</options>
+  </rule>
+
+  <rule id="20434" level="3">
```
Comment (chemamartinez) [inline rules/0575-windows_rules.xml]: Level 5

File: rules/0575-windows_rules.xml [hunk 31]
```diff
@@ -0,0 +1,5170 @@
...
+    <if_sid>20421</if_sid>
+    <field name="win.system.eventID">^3091$</field>
+    <description>Cannot read checkpoint record</description>
+    <options>no_full_log</options>
+  </rule>
+
+  <rule id="20435" level="3">
```
Comment (chemamartinez) [inline rules/0575-windows_rules.xml]: Level 5

File: rules/0575-windows_rules.xml [hunk 32]
```diff
@@ -0,0 +1,5170 @@
...
+    <if_sid>20421</if_sid>
+    <field name="win.system.eventID">^3099$</field>
+    <description>Unable to terminate notifications normally</description>
+    <options>no_full_log</options>
+  </rule>
+
+  <rule id="20436" level="3">
```
Comment (chemamartinez) [inline rules/0575-windows_rules.xml]: Level 5

File: rules/0575-windows_rules.xml [hunk 33]
```diff
@@ -0,0 +1,5170 @@
...
+    <if_sid>20421</if_sid>
+    <field name="win.system.eventID">^7011$</field>
+    <description>Directory location invalid</description>
+    <options>no_full_log</options>
+  </rule>
+
+  <rule id="20437" level="4">
```
Comment (chemamartinez) [inline rules/0575-windows_rules.xml]: Level 5

File: rules/0575-windows_rules.xml [hunk 34]
```diff
@@ -0,0 +1,5170 @@
...
+    <if_sid>20421</if_sid>
+    <field name="win.system.eventID">^7013$</field>
+    <description>Disk full, update paused</description>
+    <options>no_full_log</options>
+  </rule>
+
+  <rule id="20438" level="3">
```
Comment (chemamartinez) [inline rules/0575-windows_rules.xml]: Level 5

File: rules/0575-windows_rules.xml [hunk 35]
```diff
@@ -0,0 +1,5170 @@
...
+  </rule>
+
+  <rule id="20439" level="4">
+    <if_sid>20421</if_sid>
+    <field name="win.system.eventID">^7066$</field>
+    <description>Missing directory configuration</description>
+    <options>no_full_log</options>
+  </rule>
+
+  <rule id="20440" level="3">
```
Comment (chemamartinez) [inline rules/0575-windows_rules.xml]: Level 5

File: rules/0575-windows_rules.xml [hunk 36]
```diff
@@ -0,0 +1,5170 @@
...
+  </rule>
+
+  <rule id="20438" level="3">
+    <if_sid>20421</if_sid>
+    <field name="win.system.eventID">^7040$</field>
+    <description>Index data cannot be read</description>
+    <options>no_full_log</options>
+  </rule>
+
+  <rule id="20439" level="4">
```
Comment (chemamartinez) [inline rules/0575-windows_rules.xml]: Level 5

File: rules/0575-windows_rules.xml [hunk 37]
```diff
@@ -0,0 +1,5170 @@
...
+    <if_sid>20421</if_sid>
+    <field name="win.system.eventID">^7064$</field>
+    <description>Performance monitoring cannot be initialized because the counters are not loaded or the shared memory object cannot be opened</description>
+    <options>no_full_log</options>
+  </rule>
+
+  <rule id="20444" level="3">
+    <if_sid>20421</if_sid>
+    <field name="win.system.eventID">^1010$</field>
+    <description>The
```
Comment (chemamartinez) [inline rules/0575-windows_rules.xml]: Level 5

Comment (chemamartinez) [inline rules/0575-windows_rules.xml]: Level 5

Comment (chemamartinez) [inline rules/0575-windows_rules.xml]: Level 5

Comment (chemamartinez) [inline rules/0575-windows_rules.xml]: Level 5

Comment (chemamartinez) [inline rules/0575-windows_rules.xml]: Level 5

Comment (chemamartinez) [inline rules/0575-windows_rules.xml]: Level 4

Comment (chemamartinez) [inline rules/0575-windows_rules.xml]: Level 7 (corrupted database)

Comment (chemamartinez) [inline rules/0575-windows_rules.xml]: Level 5

Comment (chemamartinez) [inline rules/0575-windows_rules.xml]: Level 5

Comment (chemamartinez) [inline rules/0575-windows_rules.xml]: Level 4

Comment (chemamartinez) [inline rules/0575-windows_rules.xml]: Level 5

Comment (chemamartinez) [inline rules/0575-windows_rules.xml]: These rules should be included in the groups `windows` and `windows_defender` as they were in the other file.

Comment (chemamartinez) [inline rules/0575-windows_rules.xml]: Group `mcafee` for these rules: `<group name="mcafee,"> ... </group>`

Comment (chemamartinez) [inline rules/0575-windows_rules.xml]: What happens​ with this rule? If it works, it should be uncommented. If not, delete it.

Comment (chemamartinez) [inline rules/0575-windows_rules.xml]: Groups `windows` and `mse`

Comment (chemamartinez) [inline rules/0575-windows_rules.xml]: Groups `windows` and `windows_logs`

Comment (chemamartinez) [inline rules/0575-windows_rules.xml]: Group `sysmon`

Comment (chemamartinez) [inline rules/0575-windows_rules.xml]: Groups `windows` and `ipsec`

Comment (cristgl) [reply rules/0575-windows_rules.xml]: This is weird, the rule is not commented but is shown as if it was. There are no comment characters (`<!--, -->`)

**Comment by chemamartinez** [comment]
Can we remove the `Windows: ` prefix from the rules description for the file `0575-windows_rules.xml`. Here an example:

```
<rule id="20008" level="4">
    <if_sid>20005</if_sid>
    <field name="win.system.eventID">^577$|^4673$</field>
    <description>Windows: Failed attempt to perform a privileged operation</description>
    <options>no_full_log</options>
    <group>authentication_success,pci_dss_10.2.2,gdpr_IV_32.2,</group>
  </rule>
```

Those prefixes don't give us extra information because we know is a Windows-related alert by many other factors. In addition, they appear just in a few rules.

Regards.

File: rules/0575-windows_rules.xml [hunk 38]
```diff
@@ -0,0 +1,5187 @@
...
+    <if_sid>20001</if_sid>
+    <field name="win.system.providerName">VSS</field>
+    <description>Group of VSS events</description>
+    <options>no_full_log</options>
+  </rule>
+
+  <rule id="20448" level="3">
+    <if_sid>20447</if_sid>
+    <field name="win.system.eventID">^8224$</field>
+    <desc
```
Comment (chemamartinez) [inline rules/0575-windows_rules.xml]: What means that `(0)`? I think the expected here is the instance ID or name

Comment (chemamartinez) [inline rules/0575-windows_rules.xml]: Same question as for rule 20578

Comment (chemamartinez) [inline rules/0575-windows_rules.xml]: Successfully -> level 3

Comment (chemamartinez) [inline rules/0575-windows_rules.xml]: Unexpected behavior -> level 5

Comment (chemamartinez) [inline rules/0575-windows_rules.xml]: Level 5 as well

Comment (chemamartinez) [inline rules/0575-windows_rules.xml]: This rule is not valid. Windows 2000 doesn't support the eventchannel format. Please review if any other rules are related to this OS.

Comment (chemamartinez) [inline rules/0575-windows_rules.xml]: The fault tolerance driver configuration information is **corrupted**

Comment (cristgl) [reply rules/0575-windows_rules.xml]: You are right, the instance and size values are variable. As they are returned in multiple `Data` fields, the description will be leaved like this: `The version store for this instance has reached its maximum size`

Comment (cristgl) [reply rules/0575-windows_rules.xml]: I got the description from this web page: https://kb.eventtracker.com/evtpass/evtPages/EventId_102_ESENT_46807.asp
But it is not correct, the real event describes that the database engine is starting a new instance: https://www.experts-exchange.com/questions/21094461/ESENT-Appearing-in-Event-log-all-of-a-sudden.html

File: rules/0575-windows_rules.xml [hunk 39]
```diff
@@ -0,0 +1,5187 @@
...
+    <field name="win.system.eventID">^200$|^300$|^302$</field>
+    <description>TS Gateway login success</description>
+    <options>no_full_log</options>
+    <group>authentication_success,pci_dss_10.2.5,gpg13_7.1,gpg13_7.2,gdpr_IV_32.2,</group>
+  </rule>
+
+  <rule id="20098" level="0">
+    <if_sid>20308</if_sid>
+    <field name="win.system.message">\.*The database engine has successfully completed recovery steps|The database engine is initiating recovery steps</field>
+    <description>Ignore rule 200157: TS Gateway login success</description>
```
[the remark this answers is not in the document]
Comment (cristgl) [reply rules/0575-windows_rules.xml]: It must be a mismatch when replacing some IDs. It is referred to 20308: `TS Gateway login success`

**Comment by cristgl** [comment]
To sort the rules, several generic groups are being created, we want to classify events by channel, if possible. For example, we would have groups like: `windows_severity` for the main rules containing the severity value of the event (information, warning, error, audit_failure, audit_success), `windows_security` for security events, `windows_application` for events coming from the Application channel and `windows_system` for each system event. This will give much more organization and ease to add new rules. At the time there are some events that we are not sure from which channel are them but we can find it out searching for the event ID.

**Comment by cristgl** [comment]
Testing process
--------------------
- [x] Add new user.
- [x] Install and uninstall an application.
- [x] Logoff and logon.
- [ ] Enable every audit policy and see if it does not flood.
- [x] See that Application alerts pop up when using the system normally.
- [ ] Change an account's permissions.


For the next Windows systems:

- Windows Server 2008
- Windows Server 2012

**Pull request merged**